mleap-spark-testkit/src/main/scala/org/apache/spark/ml/parity/SparkParityBase.scala

```diff
@@ -131,10 +136,45 @@ abstract class SparkParityBase extends FunSpec with BeforeAndAfterAll {
     }
   }
 
+  def checkRowWithRelTol(actualAnswer: Row, expectedAnswer: Row, eps: Double): Unit = {
+    assert(actualAnswer.length == expectedAnswer.length,
+      s"actual answer length ${actualAnswer.length} != " +
+        s"expected answer length ${expectedAnswer.length}")
+
+    actualAnswer.toSeq.zip(expectedAnswer.toSeq).foreach {
+      case (actual: Double, expected: Double) =>
+        assert(actual ~== expected relTol eps)
+      case (actual: Float, expected: Float) =>
+        assert(actual ~== expected relTol eps)
+      case (actual: Vector, expected: Vector) =>
+        assert(actual ~= expected relTol eps)
+      case (actual: Seq[_], expected: Seq[_]) =>
+        assert(actual.length == expected.length, s"actual length ${actual.length} != " +
+          s"expected length ${expected.length}")
+        actual.zip(expected).foreach {
+          case (actualElem: Double, expectedElem: Double) =>
+            assert(actualElem ~== expectedElem relTol eps)
+          case (actualElem: Float, expectedElem: Float) =>
+            assert(actualElem ~== expectedElem relTol eps)
+          case (actualElem, expectedElem) =>
+            assert(actualElem == expectedElem, s"$actualElem did not equal $expectedElem")
+        }
+      case (actual: Row, expected: Row) =>
+        checkRowWithRelTol(actual, expected, eps)
+      case (actual, expected) =>
+        assert(actual == expected, s"$actual did not equal $expected")
+    }
+  }
+
+  var relTolEps: Double = 1E-3
```

Comment:
I agree that comparing floats/doubles for exact equality could lead to issues but I think the order of magnitude here is too big, perhaps something like 1E-9 or 1E-12?